src/value.c

```diff
@@ -392,31 +392,27 @@ int SIArray_Compare(SIValue arrayA, SIValue arrayB) {
 		int compareResult = SIValue_Compare(aValue, bValue);
 		switch(compareResult) {
 		case 0:
-			// we have a matching value
+			// We have a matching value.
 			break;
 		case COMPARED_NULL:
-			// there was a null comparison
-			nullCompare++;  // update null comparison counter
-			allDisjoint++;  // null comparison is also a disjoint comparison
-			break;
-		case DISJOINT:
-			allDisjoint++;  // there was a disjoint comparison
-			if(notEqual == 0) notEqual =
-					compareResult; // if there wasn't any false comparison, update the false comparison to disjoint value
+			// There was a null comparison.
+			nullCompare++;  // Update null comparison counter.
+			notEqualCounter++;  // Null comparison is also a false comparison.
 			break;
 		default:
-			// if comparison is not 0 (a != b), set the first value
+			// If comparison is not 0 (a != b), set the first value.
+			notEqualCounter++;
 			if(notEqual == 0) notEqual = compareResult;
 			break;
 		}
```

Comment:
Isn't there a situation in which we can break out of this loop early?
Do we need to scan through the entire array? (minLengh) ?